spec/support/custom_matchers.rb

```diff
@@ -0,0 +1,36 @@
+RSpec::Matchers.define :have_message do |expected|
+  match do |actual|
+    actual_messages = Array(actual).map { |m| strip_html(m.to_s) }
+    actual_messages.any? { |m| m.strip == strip_html(expected).strip }
+  end
+
+  failure_message do |actual|
+    actual_stripped = Array(actual).map { |m| strip_html(m.to_s) }
```

Comment:
We should wrap this block (and the similar one in `:contain_message` in a [`:nocov:`](https://github.com/simplecov-ruby/simplecov?tab=readme-ov-file#ignoringskipping-code) so that it isn't included in the coverage report.